### PR TITLE
WIP Ceph: add and clarify best practices in docs and manifests 

### PR DESCRIPTION
**Pull request by BlaineEXE**
**Description of your changes:**

Add and clarify best practices in Rook docs and manifests.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[skip ci]

### CODE REVIEW

File: cluster/examples/kubernetes/ceph/cluster.yaml [hunk 1]
```diff
@@ -79,8 +79,10 @@ spec:
     disable: false
   # To control where various services will be scheduled by kubernetes, use the placement configuration sections below.
   # The example under 'all' would have all services scheduled on kubernetes nodes labeled with 'role=storage-node' and
-  # tolerate taints with a key of 'storage-node'.
+  # tolerate taints with a key of 'storage-node'. Individual daemon placements are merged with `all`; they do not
+  # override `all`.
 #  placement:
+#    #
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster.yaml]: is this blank line intentional?

File: cluster/examples/kubernetes/ceph/operator.yaml [hunk 2]
```diff
@@ -33,9 +35,12 @@ data:
   #     key: node-role.kubernetes.io/etcd
   #     operator: Exists
   # # (Optional) Ceph CSI plugin NodeAffinity.
-  # CSI_PLUGIN_NODE_AFFINITY: "role=storage-node; storage=rook, ceph"
-  # # (Optional) CEPH CSI plugin tolerations list. Put here list of taints you want to tolerate in YAML format.
+  # # (Optional) Ceph Provisioner NodeAffinity. e.g., role=storage-node; storage=rook,ceph
```
Comment (umangachapagain) [inline cluster/examples/kubernetes/ceph/operator.yaml]:
```suggestion
  # # (Optional) Ceph CSI plugin NodeAffinity. e.g., role=storage-node; storage=rook,ceph
```

File: cluster/examples/kubernetes/ceph/operator.yaml [hunk 3]
```diff
@@ -33,9 +35,12 @@ data:
   #     key: node-role.kubernetes.io/etcd
   #     operator: Exists
   # # (Optional) Ceph CSI plugin NodeAffinity.
```
Comment (umangachapagain) [inline cluster/examples/kubernetes/ceph/operator.yaml]: Remove this line.

File: cluster/examples/kubernetes/ceph/cluster.yaml (hunk 1 above)
Comment (BlaineEXE) [reply cluster/examples/kubernetes/ceph/cluster.yaml]: Nope. Fixed.

File: cluster/examples/kubernetes/ceph/operator.yaml [hunk 4]
```diff
@@ -21,21 +21,25 @@ metadata:
   # should be in the namespace of the operator
   namespace: rook-ceph
 data:
-  # # (Optional) Ceph Provisioner NodeAffinity.
-  # CSI_PROVISIONER_NODE_AFFINITY: "role=storage-node; storage=rook, ceph"
-  # # (Optional) CEPH CSI provisioner tolerations list. Put here list of taints you want to tolerate in YAML format.
+  # # (Optional) Ceph Provisioner NodeAffinity. e.g., role=storage-node; storage=rook,ceph
+  # # Node affinities are added in <key>=<value>[,<value>]... syntax separated by semicolons (;)
+  # # Affinities are added to CSI pods with `requiredDuringSchedulingIgnoredDuringExecution`
```
Comment (Madhu-1) [inline cluster/examples/kubernetes/ceph/operator.yaml]: Can we change CSI pods to CSI provisioner pods. As this applies only to provisioner pod

File: cluster/examples/kubernetes/ceph/operator.yaml [hunk 5]
```diff
@@ -21,21 +21,25 @@ metadata:
...
   #     key: node-role.kubernetes.io/controlplane
   #     operator: Exists
   #   - effect: NoExecute
   #     key: node-role.kubernetes.io/etcd
   #     operator: Exists
-  # # (Optional) Ceph CSI plugin NodeAffinity.
-  # CSI_PLUGIN_NODE_AFFINITY: "role=storage-node; storage=rook, ceph"
-  # # (Optional) CEPH CSI plugin tolerations list. Put here list of taints you want to tolerate in YAML format.
+  # # (Optional) Ceph CSI plugin NodeAffinity. e.g., role=storage-node; storage=rook,ceph
+  # # Node affinities are added in <key>=<value>[,<value>]... syntax separated by semicolons (;)
```
Comment (Madhu-1) [inline cluster/examples/kubernetes/ceph/operator.yaml]: Same here also CSI pods to CSI daemonset or plugin pods

**Comment by leseb** [comment]
@BlaineEXE any plan to pursue this work? Thanks

**Pull request closed without merging**